onnxruntime/contrib_ops/cpu/attnlstm/bahdanau_attention.cc

```diff
@@ -63,11 +63,11 @@ template <typename T>
 void BahdanauAttention<T>::PrepareMemory(
     const gsl::span<const T>& memory,
     const gsl::span<const int>& memory_sequence_lengths) {
-  std::copy(memory.cbegin(), memory.cend(), values_.begin());
```

Comment:
This is why I think the migration to MS GSL the sooner the better. In the future, we may consider to create a safer wrapper of std::copy (and memcpy) with range check. Here when we convert spans to iterators, I believe the range information is lost, and the main benefit of using span is gone.